SmartTutorTests/Unit/ContentServiceTests.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using Moq;
+using SmartTutor.ContentModel;
+using SmartTutor.ContentModel.LectureModel;
+using SmartTutor.ContentModel.LectureModel.Repository;
+using SmartTutor.ContentModel.ProgressModel;
+using SmartTutor.ContentModel.ProgressModel.Repository;
+using SmartTutor.Recommenders;
+using Xunit;
+
+namespace SmartTutorTests.Unit
+{
+    public class ContentServiceTests
+    {
+        private readonly IContentService _contentService;
+        private readonly Mock<IRecommender> _recommender = new Mock<IRecommender>();
+
+        public ContentServiceTests()
+        {
+            _contentService = new ContentService(_recommender.Object, CreateMockLectureRepository(), null,
+                CreateMockTraineeRepository());
+        }
+
+        private static ITraineeRepository CreateMockTraineeRepository()
+        {
+            Mock<ITraineeRepository> traineeRepo = new Mock<ITraineeRepository>();
+            traineeRepo.Setup(repo => repo.GetNodeProgressForTrainee(1, 1)).Returns(new NodeProgress());
+            traineeRepo.Setup(repo => repo.GetTraineeById(1)).Returns(Trainee);
+            return traineeRepo.Object;
+        }
+
+        private static ILectureRepository CreateMockLectureRepository()
+        {
+            Mock<ILectureRepository> lectureRepo = new Mock<ILectureRepository>();
+            lectureRepo.Setup(repo => repo.GetKnowledgeNodeWithSummaries(2)).Returns(KnowledgeNode);
+            return lectureRepo.Object;
+        }
+
+        [Theory]
+        [MemberData(nameof(TraineeTestData))]
+        public void Creates_node_content(int nodeId, int traineeId, int numOfInvocations)
+        {
+            _contentService.GetNodeContent(nodeId, traineeId);
+            _recommender.Verify(recommender => recommender.BuildNodeProgressForTrainee(Trainee, KnowledgeNode),
+                Times.Exactly(numOfInvocations));
+        }
+
+        public static IEnumerable<object[]> TraineeTestData =>
+            new List<object[]>
+            {
+                new object[]
+                {
+                    1,
+                    1,
+                    0
```

Comment:
This test is too coupled to the implementation and fails to satisfy the property of a good test "resistant to refactoring".

We should be verifying if we are getting a new node or an existing node based on the trainee/node id combination. How that's implemented is of little concern.

Basically, the MockRepo should return an existing node progress (e.g., with ID 1) while the creation function should make a new one (e.g., ID 7).